pinot-minion/src/main/java/org/apache/pinot/minion/taskfactory/TaskFactoryRegistry.java

```diff
@@ -108,6 +120,8 @@ public TaskResult run() {
                 minionMetrics.addMeteredTableValue(taskType, MinionMeter.NUMBER_TASKS_FAILED, 1L);
                 LOGGER.error("Caught exception while executing task: {}", _taskConfig.getId(), e);
                 return new TaskResult(TaskResult.Status.FAILED, e.toString());
+              } finally {
+                minionMetrics.addValueToGlobalGauge(MinionGauge.NUMBER_OF_TASKS, -1L);
```

Comment:
Best to call a method `runInternal` that has the current logic, and manipulate this metric before and after the call outside (in the run() method). That way, if someone adds a return from an earlier point, we don't have to worry about the metric going bad.

Reply:
Done.